functions/index.js

```diff
@@ -35,27 +40,58 @@ exports.createNewTask = functions.https.onRequest((request, response) => {
         console.log(category);
         console.log(createNewTaskSprintNumber);
 
-        db.collection(fullSprintId).doc(category).set({
-                Title: title,
-                Description: des,
-                Priority: priority,
-                Difficulty: difficulty,
-                Creator: creator,
-                Assignee: assignee,
-                ET: estimatedTime,
-                Status: status,
-                Category: category,
-                LogWorkTotalTime: loggedWorkTotalTime,
-                WorkDone: workDone,
-                CreateNewTaskSprintNumber: createNewTaskSprintNumber
+        db.collection("Main").doc("RawData").get().then((doc) => {
+
+                totalDevelopmentTask = doc.data().TotalDevelopmentTask + 1;
+                totalBusinessTask = doc.data().TotalBusinessTask + 1;
+                totalMarketingTask = doc.data().TotalBusinessTask + 1;
+                totalNumberOfTask = (totalDevelopmentTask + totalBusinessTask + totalMarketingTask);
+
+                if (category === "Development") {
+                    taskId = category[0] + totalDevelopmentTask;
+                } else if (category === "Business") {
+                    taskId = category[0] + totalBusinessTask;
+                } else {
+                    taskId = category[0] + totalMarketingTask;
+                }
+
+                console.log(taskId);
+
+                var setDataPromise = db.collection(fullSprintId).doc(taskId).set({
+                    Title: title,
+                    Description: des,
+                    Priority: priority,
+                    Difficulty: difficulty,
+                    Creator: creator,
+                    Assignee: assignee,
+                    ET: estimatedTime,
+                    Status: status,
+                    Category: category,
+                    LogWorkTotalTime: loggedWorkTotalTime,
+                    WorkDone: workDone,
+                    CreateNewTaskSprintNumber: createNewTaskSprintNumber
+                });
+                return Promise.resolve(setDataPromise);
             })
-            .then(() => {
-                var work = { data: "working" }
+            .then(function(setDataPromise) {
+                var updateSetDataPromise = db.collection("Main").doc("RawData").update({
+                    TotalDevelopmentTask: totalDevelopmentTask,
+                    TotalBusinessTask: totalBusinessTask,
+                    TotalMarketingTask: totalMarketingTask,
+                    TotalNumberOfTask: totalNumberOfTask
```

Comment:
And those new Counters can also be updated here

Reply:
okay

Reply:
Two more counters will be present in db --> "Main" --> "RawData" 
1. TotalCompletedTask
2. TotalUncompletedTask

Please Update these counters also here.

Reply:
ok